src/lv_message_efficient.cc

```diff
@@ -67,25 +67,17 @@ namespace grpc_labview
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    const char *LVMessageEfficient::ParseString(google::protobuf::uint32 tag, const MessageElementMetadata& fieldInfo, uint32_t index, const char *protobuf_ptr, ParseContext *ctx)
-    {  
+    const char* LVMessageEfficient::ParseString(google::protobuf::uint32 tag, const MessageElementMetadata& fieldInfo, uint32_t index, const char* protobuf_ptr, ParseContext* ctx)
+    {
         const char* lv_ptr = (this->GetLVClusterHandleSharedPtr()) + fieldInfo.clusterOffset;
 
         if (fieldInfo.isRepeated)
-        {            
-            std::string key = fieldInfo.fieldName +  std::to_string(_currentIndexForRepeatedMessageValue);
-         
-            // Get the _repeatedMessageValues vector from the map
-            auto _repeatedStringValuesIt = _repeatedStringValuesMap.find(key);
-            if (_repeatedStringValuesIt == _repeatedStringValuesMap.end())
-            {
-                _repeatedStringValuesIt = _repeatedStringValuesMap.emplace(key, google::protobuf::RepeatedField<std::string>()).first;
-            }
-
+        {
+            auto repeatedString = google::protobuf::RepeatedField<std::string>();
```

Comment:
NOTE: There is no reason to hold onto this repeated string field beyond this stack frame. I've removed the backing fields from the message that were being used to track it and just allocate everything on the stack now for efficiency/clarity.

Reply:
The reason we had to maintain the lifetime of the repeated field longer than the stack was for much larger messages. When we do have very large messages, gRPC breaks the data into smaller chunks in memory, and parsing the message happens with multiple calls to `LVMessage::_InternalParse`, and not on a single call.
Could you try running tests with large repeated messages?

Reply:
I'm not sure I follow. What would be considered a large repeated message for this case? I assume you are talking about any message with a repeated string field of sufficiently large length? I think the wire protocol technically allows for the same field to be sent multiple times where last one wins for scalar fields and values are appended for repeated fields. Is that what you are referring to? If so, I might have to think on that some more. I'm not really sure how to test that without writing the byte stream directly.

Reply:
I think I see what you are talking about. I guess in the case of repeated strings, we were potentially doing multiple copies into the LV array rather than one copy from the repeated field into the LV array at the very end. I find reusing the message on the stack makes it really difficult to reason about.

Reply:
The latest changes revert back to something closer to the original. I've changed the implementation so repeated strings and messages more or less work the same so hopefully the code is easier to read and reason about. For this to work in a more straight forward way, I now allocate a new nested message on the stack for each element in the repeated field rather than reusing it and dealing with state from the previous element polluting the data for the next element.